packages/js-moi-logic/src/logic-factory.ts

```diff
@@ -102,15 +102,15 @@ export class LogicFactory extends LogicBase {
 
         if(builder) {
             const builderRoutine = builder.data as LogicManifest.Routine;
-
-            if(builderRoutine.accepts && Object.keys(builderRoutine.accepts).length != args.length) {
+            
+            if(builderRoutine.accepts && (args.length < Object.keys(builderRoutine.accepts).length)) {
```

Comment:
don't not include optional params while checking for the expected param length.

Reply:
a148b25d7be6dd2e55ef9f4502f56ebc49018142